modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/components/form-renderer-custom-fields/RequiredField.es.js

```diff
@@ -212,10 +212,13 @@ export default ({AppContext, dataLayoutBuilder}) => {
 					onToggle={(toggle) => {
 						setToggle(toggle);
 
-						dataLayoutBuilder.dispatch('fieldEdited', {
-							propertyName: 'required',
-							propertyValue: toggle,
-						});
+						dataLayoutBuilder.formBuilderWithLayoutProvider.refs.layoutProvider?.dispatch?.(
```

Comment:
For avoiding accessing those refs, What do you think about left `dispatch`, `getState`, `getRules` methods on the class?